lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -1374,6 +1388,28 @@ MemOp::PortKind MemOp::getPortKind(size_t resultNo) {
       getResult(resultNo).getType().cast<FIRRTLType>());
 }
 
+/// Return the number of bits in the mask for the memory.
+size_t MemOp::getMaskBits() {
+
+  for (auto res : getResults()) {
+    auto firstPortType = res.getType().cast<FIRRTLType>();
+    if (getMemPortKindFromType(firstPortType) == PortKind::Read)
+      continue;
+
+    FIRRTLType mType;
+    for (auto t :
+         firstPortType.getPassiveType().cast<BundleType>().getElements()) {
+      if (t.name.getValue().contains("mask"))
+        mType = t.type;
+    }
+    if (mType.dyn_cast_or_null<UIntType>())
+      return mType.getBitWidthOrSentinel();
+  }
+  // Mask of zero bits means, either there are no write/readwrite ports or the
+  // mask is of aggregate type.
```

Comment:
This might be good to move into the main documentation comment for the method, since it's an important thing to know about the return value.

Reply:
Good idea, a proper documentation for `firrtl.mem` is missing. I will add it in a subsequent PR.